mods/persistence/modules/modular_computers/file_system/programs/trade_beacon_program.dm

```diff
@@ -0,0 +1,560 @@
+/datum/computer_file/program/trade_management
+	filename = "trademnrger"
+	filedesc = "Import/Export Management"
+	extended_desc = "This program allows you to process imports or exports from nearby trade beacons."
+
+	program_icon_state = "generic"
+	program_key_state = "generic_key"
+	program_menu_icon = "cart"
+
+	available_on_network = 1
+
+	size = 12
+	nanomodule_path = /datum/nano_module/program/trade_management
+	category = PROG_SUPPLY
+
+	var/obj/effect/overmap/trade_beacon/selected_beacon
+
+	var/import_menu = 1
+	var/export_menu = 0
+	var/log_menu = 0
+	var/datum/beacon_import/selected_import
+	var/datum/beacon_export/selected_export
+	var/payment_type = 1 // 1 == personal account 2 == chargestick 3 == network parent account
+
+	var/list/detected_telepads = list()
+	var/obj/selected_telepad
+
+/datum/computer_file/program/trade_management/proc/detect_telepads()
+	detected_telepads = list()
+	for(var/obj/machinery/telepad_cargo/c in orange(3, get_turf(computer)))
```

Comment:
Not necessary, but you could always use networks to get your connected telepad instead of doing a physical search.

Reply:
That's true, but I wanted importing/exporting to be limited to certain areas (telepad rooms) and the possibility of public shipping areas. Unless it'a specific requirement, I'd like to keep the system as is.

Reply:
It's all good.